drivers/ukintctlr/gic/include/uk/intctlr/gic.h

```diff
@@ -86,15 +86,16 @@ struct _gic_operations {
 	/** Disable IRQ */
 	void (*disable_irq)(uint32_t irq);
 	/** Set IRQ trigger type */
-	void (*set_irq_type)(uint32_t irq, enum uk_irq_trigger trigger);
+	void (*set_irq_trigger)(uint32_t irq,
+				enum uk_intctlr_irq_trigger trigger);
 	/** Set priority for IRQ */
 	void (*set_irq_prio)(uint32_t irq, uint8_t priority);
 	/** Set IRQ affinity (or "target" for GICv2) */
 	void (*set_irq_affinity)(uint32_t irq, uint32_t affinity);
 	/** Translate to absolute IRQ according to type e.g. PPI SPI SGI */
 	uint32_t (*irq_translate)(uint32_t type, uint32_t irq);
 	/** Handle IRQ */
-	void (*handle_irq)(struct __regs *regs);
+	void (*handle_irq)(struct __regs *regs, uint32_t __unused irq);
```

Comment:
It's because `uk_intctlr->ops->handle_irq()` has such a parameter because of x86 that requires passing the interrupt number to the handler. An alternative would be to keep this pure and introduce a wrapper in `gic/ukintctlr.c` that ignores the `irq` parameter and calls this function. Perhaps that's more clean, what do you think?

Reply:
Not only we ignore it, but we also populate it with a dummy value. That is because on arm you retrieve (and ack) an interrupt when you read `GICC_IAR`, but at the same time you are expected to loop until you read the spurious IRQ (see the implementation in either gic-v2 or gic-v3). Because of that, we cannot move the interrupt retrieval logic outside `gic->handle_irq`. These are the merits of unified interrupt handing ops 👹

Reply:
So my understanding is that the callchain looks like this:
`trap_el1_irq(regs)` => `gic->ops.handle_irq(regs, 0)` = `gicv3_handle_irq(regs, 0)` => `uk_intctlr_irq_handle(regs, irq)`

That's why I don't see a need for having the irq parameter before `uk_intctlr_irq_handle`.

Reply:
Sorry I see where is the gap now. I was supposed to update `trap_el1_irq` to call the generic `handle_irq` from the `uk_intctlr` ops, as a step towards gradually moving all the calls to the driver via the API and deprecate direct access to driver ops.

Reply:
Update: I removed it again, see https://github.com/unikraft/unikraft/pull/1103#issuecomment-1735547936